pkg/arvo/app/btc-node-store.hoon

```diff
@@ -0,0 +1,146 @@
+::  btc-node-store: data store for state received from a bitcoin full node
+::
+/-  *btc-node-store
+/+  *btc-node-json, default-agent, verb
+::
+=>  |%
+    ::
+    +$  card  card:agent:gall
+    ::
+    +$  state
+      $%  [%0 state-zero]
+      ==
+    ::
+    +$  state-zero
+      $:  =wallets
+          default-wallet=@t
+      ==
+    --
+::
+=|  state-zero
+=*  state  -
+::  Main
+::
+%+  verb  |
+^-  agent:gall
+=<  |_  =bowl:gall
+    +*  this      .
+        btc-core  +>
+        bc        ~(. btc-core bowl)
+        def       ~(. (default-agent this %|) bowl)
+    ::
+    ++  on-init
+      ^-  (quip card _this)
+      :-  ~
+      %=    this
+        wallets  (~(put by wallets) [*@t *wallet])
+      ==
+    ::
+    ++  on-save   !>(state)
+    ++  on-load
+      |=  old=vase
+      `this(state !<(state-zero old))
+    ::
+    ++  on-poke
+      |=  [=mark =vase]
+      ^-  (quip card _this)
+      ~&  got-poked-with-data=mark
+      |^
+      ?+    mark    (on-poke:def mark vase)
+          %btc-node-store-action
+        (store-action !<(btc-node-store-action vase))
+      ::
+          %btc-node-store-command
+        (store-command !<(btc-node-store-command vase))
+      ==
+      ::
+      ++  store-action
+        |=  action=btc-node-store-action
+        ^-  (quip card _this)
+        =^  cards  state
+          ?+  -.action  ~|  [%unsupported-action -.action]  !!
+            %add-wallet     (handle-add:bc +.action)
+            %list-wallets   handle-list-wallet:bc
+            %update-wallet  (handle-update-wallet:bc +.action)
+          ==
+        [cards this]
+      ::
+      ++  store-command
+        |=  command=btc-node-store-command
+        ^-  (quip card _this)
+        =^  cards  state
+          ?+  -.command  ~|  [%unsupported-command -.command]  !!
+              %switch-wallet  (handle-switch:bc +.command)
+          ==
+        [cards this]
+      --
+    ++  on-watch  on-watch:def
+    ++  on-leave  on-leave:def
+    ::  +on-peek: read from app state
+    ::
+    ::    default-wallet  .^(@t %gx /=btc-node-store=/default-wallet/noun)
+    ::    n-wallets       .^(@ud %gx /=btc-node-store=/n-wallets/noun)
```

Comment:
Interface documentation, sweet! I'd recommend putting this at the top of the file (alongside more info on how to use this, optionally) for discoverability reasons, but we don't really have any standards around this yet.